### DISCUSSION TITLE
Kanban to Slides

### DISCUSSION THREAD

**Discussion by matthewturk** [ebullient/obsidian-slides-extended#70] [Show and tell]
I have *forever* wanted to use a Kanban interface for creating talks, and with `slides-extended` and the Kanban plugin I finally have it!  I'm going to write it up properly in a blog post, but here's a templater template that I use to do it.  If you have a column called "Slides" it will take everything in it and export.  (It checks when you create the new template if it should.)  Here's my full template file, frontmatter included.  I am pretty excited about using this!

**Edit**: I've written it up, with a picture, [on my blog](https://matthewturk.github.io/post/kanban-slides-in-obsidian/).

```
---
aliases: []
tags:
  - talks/mine
created: 2024-11-30
modified: 2024-11-30
theme: mturk-slides
transition: none
center: false
fragments: true
hash: true
width: 1024
height: 768
margin: 0.02
defaultTemplate: "[[blankslide]]"
backgroundTransition: none
reveald3:
  - runLastState: false
enableTitleSlide: false
title: TITLE GOES HERE
---

<%*
const dv = tp.app.plugins.getPlugin("dataview").api;
const filePath = tp.config.active_file;
const converted = [];
if (filePath) {
	const p = dv.page(filePath.path)
	const f = p.file;
	if (p['kanban-plugin'] == 'board') {
		console.log("hi");
		const items = f.tasks.filter(t => t.section.subpath == "Slides");
		items.forEach( i => {
			if (i.text.startsWith("[[") && i.text.endsWith("]]")) {
				converted.push(`!${i.text}`);
			} else {
				converted.push(i.text);
			}
		});
	}
}
-%>

<% converted.join("\n\n---\n\n") -%>
```

**Comment by UUoocl** [comment]
This is a great idea!  Thanks for sharing the template.

**Comment by UUoocl** [comment]
@matthewturk Thanks again for sharing how to organize slides with the Kanban plugin.  It's inspired these additions. 

### Kanban board template

This template will create a Kanban Board with cards for each note that has a 'Reveal-Section' tag in the Front Matter.  
Create a new note in the Templater template folder with this code. 

<pre><code>
---

kanban-plugin: board

---

## Slides


## Candidate Slides

{presentation_sections}

<%*
const dv = tp.app.plugins.getPlugin("dataview").api;
let sections="";

//create a card for each note with a 'Reveal-Section' tag
for(let page of dv.pages("#Reveal-Section").file.name){
	sections += `- [ ] [[${page}]]
`
}
tR = tR.replace('{presentation_sections}',sections);

-%>


%% kanban:settings
```
{"kanban-plugin":"board","list-collapse":[false,false]}
```
%%
</code></pre>

After creating a new note with this template, click the reading view icon to switch to the Kanban interface. 
![image](https://github.com/user-attachments/assets/cd98bf77-2299-4bf7-a100-138253499be2)

### Kanban to Presentation template:  Copy Note Content
This version of your idea adds 2 features

1. copies the content of the Section note into the New Presentation.  
2. After creating the New Presentation, the Templater command "Replace templates in the active file" is run. This allows results from the Dataview plugin to be included in the New Presentation.  

```
---
aliases:
  - "[]"
tags:
  - presentation
created: 2023-09-13
modified: 2023-09-13
theme: black
transition: none
center: true
fragments: true
hash: true
width: 1024
height: 768
margin: 0.02
defaultTemplate: 
backgroundTransition: none
bg: transparent
reveald3:
  - runLastState: false
enableTitleSlide: false
title: TITLE GOES HERE
presenterName: PRESENTER NAME
venueName: Unknown
talkDate: Unknown
---

<%*
const dv = tp.app.plugins.getPlugin("dataview").api;
const filePath = tp.config.active_file;
if (filePath) {
	const p = dv.page(filePath.path)
	const f = p.file;
	//If the Active page is a Kanban board
	if (p['kanban-plugin'] == 'board') {
		const items = f.tasks.filter(t => t.section.subpath == "Slides");
		for(i of items) {
			if (i.text.startsWith("[[") && i.text.endsWith("]]")) {
				//Remove the brackets around the note name
				const noteName = i.text.replace("[[","").replace("]]","")
				//get the note content 
				const noteTF = tp.file.find_tfile(noteName) 
				const noteContent = await app.vault.read(noteTF);
				//Remove frontmatter
				const noteContent_NoFrontMatter = noteContent.replace(/^---.*?\n---\n/s, "")
				tR += noteContent_NoFrontMatter;

			} else {
				tR += tp.file.find_tfile(i.text)
			}
		tR += "\n\n---\n\n"
		};

		tp.hooks.on_all_templates_executed(async () => {
			app.commands.executeCommandById('templater-obsidian:replace-in-file-templater');
		})
	}
}
-%>
```  

### Add Dataview Queries to slides
I want to include dataview queries in my slides, but Slides Extended doesn't process Dataview queries directly.  
Wrapping the dataview query in Templater solves this issue.  

When the "Kanban to Presentation" template runs, the dataview queries are replaced with the query results 

In a slide add a {unique placeholder} that will be replaced.  Use a different placeholder for each query.  

```
{Unique Placeholder}

<!-- <%*
dv = tp.app.plugins.getPlugin("dataview").api;
query = await dv.pages().length;
tR = tR.replace('{Unique Placeholder}',query);
%> -->
```

Your Kanban idea helped move my project forward.  Thanks for the inspiration.

**Comment by ebullient** [comment]
This is lovely! Thank you both for sharing.

**Comment by major-sam** [comment]
mermaid also has kanban implementation, but not working in slides, only obsidian 😢

**Reply by ebullient** [reply]
I think this is due to a significantly back-level version of mermaid. I can't focus on that right at the moment, if anyone wants to try to build this (see the dependabot PR for the mermaid version), that would be great.

I don't want to just merge it because it is a huge version bump.
